detox/src/artifacts/log/android/ADBLogcatRecording.js

```diff
@@ -26,26 +25,18 @@ class ADBLogcatRecording extends Artifact {
     this._waitUntilLogFileIsCreated = null;
   }
 
-  get hasRecordedFile() {
-    return !!this._waitUntilLogFileIsCreated;
-  }
-
   async doStart() {
     const pid = this.pid.get();
 
-    if (pid > 0) {
-      this.processPromise = this.adb.logcat(this.deviceId, {
-        file: this.pathToLogOnDevice,
-        time: this.since.get(),
-        pid,
-      });
+    this.processPromise = this.adb.logcat(this.deviceId, {
+      file: this.pathToLogOnDevice,
+      time: this.since.get(),
+      pid: pid > 0 ? pid : 0,
+    });
 
-      this._waitUntilLogFileIsCreated = sleep(300).then(() => {
-        return retry(() => this._assertLogIsCreated());
-      });
-    } else {
-      log.debug('Ignoring a command to start recording, because PID of the app is missing');
-    }
+    this._waitUntilLogFileIsCreated = sleep(300).then(() => {
```

Comment:
Can't we observe this event without sleeping (or shorter retry intervals with timeout)?

Reply:
No.